staging/src/k8s.io/code-generator/cmd/validation-gen/validators/skipunimported.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validators
+
+import "k8s.io/apimachinery/pkg/util/sets"
+
+const (
+	skipUnimportedTagName = "k8s:skipUnimported"
+)
+
+type skipUnimported struct{}
+
+func init() {
+	RegisterTagValidator(skipUnimported{})
+}
+
+func (skipUnimported) Init(Config) {}
+
+func (skipUnimported) TagName() string {
+	return skipUnimportedTagName
+}
+
+func (skipUnimported) ValidScopes() sets.Set[Scope] {
+	return sets.New(ScopeField)
```

Comment:
What about typedefs to non-included types and list/map values?  I think I would just do "all" here

Reply:
This is turning out to be more challenging to support.  I'm working through the logic, but it's twisty.